src/components/Chat/BottomSheetGif.js

```diff
@@ -0,0 +1,176 @@
+import IconClear from '../../assets/icon/IconClear';
+import {getGifFeatured} from '../../service/gif';
+
+const {height: heightScreen, width: widthScreen} = Dimensions.get('window');
+
+const BottomSheetGif = React.forwardRef((props, ref) => {
+  const {onCancel, onSelect} = props;
+  const [searchText, setSearchText] = React.useState('');
+  const [list, setList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+  };
+
+  const handleOnClearText = () => {
+    setSearchText('');
+  };
+
+  const onLoad = async () => {
+    setIsLoading(true);
+    try {
+      const result = await getGifFeatured();
+      if (result.code === 200) {
+        setList(result.data);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    setIsLoading(false);
+  };
+
+  React.useEffect(() => {
+    onLoad();
+  }, []);
+
+  return (
+    <BottomSheet
+      ref={ref}
+      closeOnPressMask={true}
+      height={heightScreen - 50}
+      viewstyle={styles.container}>
+      <View style={styles.searchContainer}>
+        <View style={styles.wrapperSearch}>
+          <View style={styles.wrapperIcon}>
+            <MemoIcSearch width={16.67} height={16.67} />
+          </View>
+          <TextInput
+            value={searchText}
+            onChangeText={handleChangeText}
+            multiline={false}
+            returnKeyType="search"
+            placeholder="Search"
+            placeholderTextColor={colors.gray1}
+            style={styles.input}
+          />
+
+          <TouchableOpacity
+            delayPressIn={0}
+            onPress={handleOnClearText}
+            style={styles.clearIconContainer}
+            android_ripple={{
+              color: colors.gray1,
+              borderless: true,
+              radius: 35
+            }}>
+            <View style={styles.wrapperDeleteIcon}>
+              <IconClear width={9} height={10} iconColor={colors.black} />
+            </View>
+          </TouchableOpacity>
+        </View>
+        <TouchableOpacity style={styles.wrapperButton} onPress={onCancel}>
+          <Text style={styles.cancelText}>Cancel</Text>
+        </TouchableOpacity>
+      </View>
+      <FlatList
+        data={list}
+        renderItem={({item, index}) => (
+          <TouchableOpacity
+            key={index}
+            style={{width: widthScreen / 2, height: widthScreen / 2}}
+            onPress={() => onSelect(item.media_formats.gif.url, item.media_formats.tinygif.url)}>
+            <FastImage
+              source={{uri: item.media_formats.tinygif.url}}
+              style={{width: '100%', height: '100%'}}
+              resizeMode="cover"
+            />
+          </TouchableOpacity>
+        )}
+        keyExtractor={(item, index) => index.toString()}
+        refreshing={isLoading}
+        onRefresh={onLoad}
+        numColumns={2}
+      />
+    </BottomSheet>
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This component seems to be doing a lot of things, which might make it difficult to maintain and test. Consider breaking it down into smaller components. For example, the search bar could be its own component, as well as the list of GIFs.
```diff
  return (
    <BottomSheet
      ref={ref}
      closeOnPressMask={true}
      height={heightScreen - 50}
      viewstyle={styles.container}>
      <SearchBar 
        value={searchText} 
        onChange={handleChangeText} 
        onClear={handleOnClearText} 
        onCancel={onCancel} 
      />
      <GifList 
        data={list} 
        isLoading={isLoading} 
        onRefresh={onLoad} 
        onSelect={onSelect} 
      />
    </BottomSheet>
  );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->